test/cli/whole-program_test.py

```diff
@@ -358,3 +359,35 @@ def test_checkclass_project_builddir_j(tmpdir):
     build_dir = os.path.join(tmpdir, 'b1')
     os.mkdir(build_dir)
     __test_checkclass_project(tmpdir, ['-j2', '--cppcheck-build-dir={}'.format(build_dir)])
+
+def __test_nullpointer_file0(extra_args):
+    args = [
+        '-q',
+        '--xml',
+        '--error-exitcode=1',
+        'whole-program/nullpointer1.cpp'
+    ]
+
+    args += extra_args
+
+    ret, stdout, stderr = cppcheck(args, cwd=__script_dir)
+    results = ET.fromstring(stderr)
+    file0 = ''
+    for e in results.findall('errors/error[@id="ctunullpointer"]'):
+        file0 = e.attrib['file0']
+    assert file0 == 'whole-program/nullpointer1.cpp'
+    assert stdout == ''
+    assert ret == 1, stdout
+
+
+def test_nullpointer_file0():
+    __test_nullpointer_file0(['-j1'])
+
+@pytest.mark.skip(reason="flaky!?")
```

Comment:
```suggestion
    assert ret == 1, , stdout if stdout else stderr
    assert stdout == ''
    assert file0 == 'whole-program/nullpointer1.cpp', stderr


def test_nullpointer_file0():
    __test_nullpointer_file0(['-j1'])

```

Reply:
Usually I would assume it is an issue with running the test in parallel but there is nothing in this test which could cause that.

But this test is build with UBSAN so it might indicate such an issue. The asserts are in a bad order so that would be obfuscated. I adjusted them so we should get more details (that needs to be applied across all the tests - there are already some initial helpers but something for later).

Reply:
The xfail is still in, so it might pass now but we might not see the error reported by the sanitizer.

Reply:
With xfail in, `Run test/cli (--cppcheck-build-dir)` fails (test succeeds). Without xfail, `Run test/cli` fails as expected.
How is this injection stuff supposed to work? I also see `extraargs` being used in the test.

Reply:
The injection stuff checks if any of the parameters in question already exists and only injects those is necessary. The `-j2` injection will not be done if `-j1` already exists. Same with `--cppcheck-build-dir` and `--executor`. So if you have to specify them explicitly they will not return the same no matter how it is called (e.g. Windows does not support `--executor=process`) and you need to provide an override.

This looks weird though. Will take a look later.